### DISCUSSION TITLE
How to change idle thruttle?

### DISCUSSION THREAD

**Discussion by EdwinMeriaux** [microsoft/AirSim#3860] [Support Q&A]
So from this document:
https://github.com/microsoft/AirSim/blob/master/AirLib/include/vehicles/multirotor/MultiRotorParams.hpp

It says Idle thruttle for simple flight is 50%. Is there any what to change that?

**Comment by EdwinMeriaux** [comment]
the drone that I would like to simulate has a particular max thrust and the drone's idle is lower than 50% max thrust. Is there any way to change this?

**Comment by rajat2004** [comment]
@ahmed-elsaharti might have a better idea regarding this, referring to his PR https://github.com/microsoft/AirSim/pull/3401

**Comment by ahmed-elsaharti** [comment]
Hey guys, 
If I remember correctly, the 50% idle thrust (I'm assuming we're talking about the simpleflight flight controller here) in my case was set by the gamepad/joystick connected to the computer via USB. ie: since i was using a USB gamepad with thumbsticks that default to the center, it pushed out a signal of 50% (ie: center). I remember the idle throttle dropping below 50% once no joysticks were connected to the computer.

Idr exactly where the non-joystick idle throttle was set but I'd assume it would be somewhere in the simpleflight directory since that parameter should be controlled by the autopilot code.

**Comment by EdwinMeriaux** [comment]
Hello!

Wait so this idle rate still holds when programming the drones with Python instead of flying with joystick?
I have gone through most of the files and I have not seen where I can change it...

**Comment by EdwinMeriaux** [comment]
@ahmed-elsaharti

**Reply by ahmed-elsaharti** [reply]
Sorry for replying late to this :(
In my testing, when a Joystick is not connected (or if enableApiControl is true) the idle throttle was lower than 50%. This is probably configurable but would need you to take a deeper dive into simpleflight (if thats the autopilot being used).
A quick [search for the word 'throttle' on the repo](https://github.com/microsoft/AirSim/search?q=throttle) reveals that [this section](https://github.com/microsoft/AirSim/blob/0c1ff8004932a76e745229a980f1269d4717d1a5/AirLib/include/vehicles/multirotor/firmwares/simple_flight/firmware/Params.hpp#L11-L15) would be a good place to start.
